src/main/java/org/jabref/gui/entryeditor/EntryEditor.java

```diff
@@ -134,60 +131,62 @@ public EntryEditor(BasePanel panel, EntryEditorPreferences preferences, FileUpda
                         fileLinker.copyFilesToFileDirAndAddToEntry(entry, files);
                     } else {
-                        LOGGER.debug("Mode MOVE"); //shift on win or no modifier
+                        // Shift on Windows or no modifier
+                        LOGGER.debug("Mode MOVE");
                         fileLinker.moveFilesToFileDirAndAddToEntry(entry, files);
                     }
                 }
 
                 if (dragDropPreferencesType == FileDragDropPreferenceType.COPY) {
-                    if (event.getTransferMode() == TransferMode.COPY) //ctrl on win, no modifier on Xubuntu
-                    {
+                    if (event.getTransferMode() == TransferMode.COPY) {
+                        // Ctrl on Windows, no modifier on Xubuntu
                         LOGGER.debug("Mode MOVE");
                         fileLinker.moveFilesToFileDirAndAddToEntry(entry, files);
-                    } else if (event.getTransferMode() == TransferMode.LINK) //alt on win
-                    {
+                    } else if (event.getTransferMode() == TransferMode.LINK) {
+                        // Alt on Windows
                         LOGGER.debug("Mode LINK");
                         fileLinker.addFilesToEntry(entry, files);
                     } else {
-                        LOGGER.debug("Mode COPY"); //shift on win or no modifier
+                        // Shift on Windows or no modifier
+                        LOGGER.debug("Mode COPY");
                         fileLinker.copyFilesToFileDirAndAddToEntry(entry, files);
                     }
                 }
 
                 if (dragDropPreferencesType == FileDragDropPreferenceType.LINK) {
-                    if (event.getTransferMode() == TransferMode.COPY) //ctrl on win, no modifier on Xubuntu
-                    {
+                    if (event.getTransferMode() == TransferMode.COPY) {
+                        // Ctrl on Windows, no modifier on Xubuntu
                         LOGGER.debug("Mode COPY");
                         fileLinker.copyFilesToFileDirAndAddToEntry(entry, files);
-                    } else if (event.getTransferMode() == TransferMode.LINK) //alt on win
-                    {
+                    } else if (event.getTransferMode() == TransferMode.LINK) {
+                        // Alt on Windows
                         LOGGER.debug("Mode MOVE");
                         fileLinker.moveFilesToFileDirAndAddToEntry(entry, files);
                     } else {
-                        LOGGER.debug("Mode LINK"); //shift on win or no modifier
+                        // Shift on Windows or no modifier
+                        LOGGER.debug("Mode LINK");
                         fileLinker.addFilesToEntry(entry, files);
                     }
                 }
             }
 
-            event.setDropCompleted(success);
+            event.setDropCompleted(success);  // Always false
```

Comment:
Ah glad you spotted this, of course the success should be set to true in each clause